### PR TITLE
Feat: integrate mongo support

### PR DESCRIPTION
**Pull request by mohdashraf010897**
**PR Type**: `Feature`

**Description**
The PR integrates support for `mongodb` backend specifically to `reactivesearch` methods.

[📔 Notion](https://www.notion.so/appbase/Reactivesearch-Mongo-DB-Support-1ef6e865b5cf4204aaae338d22afa288)

[🎥  Demo: Self Review](https://www.loom.com/share/b21afd8ae66546109dc3a40b220896e0)

**Related PR(s)**
- https://github.com/appbaseio/reactivesearch/pull/1810
- https://github.com/appbaseio/reactivecore/pull/111

### CODE REVIEW

File: src/core/api/reactiveSearchApi.js [hunk 1]
```diff
@@ -24,13 +26,16 @@ function reactiveSearchApi(query, settings) {
...
+  }
 
   return this.performFetchRequest({
     method: 'POST',
     path: '_reactivesearch',
     body,
-    headers: getTelemetryHeaders(this.enableTelemetry),
-    isRSAPI: true,
+    headers: getTelemetryHeaders(this.enableTelemetry, !this.mongodb),
+    isRSAPI: !this.mongodb,
```
Comment (bietkul) [inline src/core/api/reactiveSearchApi.js]: @mohdashraf010897 How is it related to mongo?

File: src/core/api/reactiveSearchv3Api.js [hunk 2]
```diff
@@ -23,14 +25,15 @@ function reactiveSearchv3Api(query, settings) {
...
+    Object.assign(body, { mongodb: getMongoRequest(this.app, this.mongodb) });
+  }
   return this.performFetchRequest({
     method: 'POST',
     path: '_reactivesearch.v3',
     body,
-    headers: getTelemetryHeaders(this.enableTelemetry),
-    isRSAPI: true,
+    headers: getTelemetryHeaders(this.enableTelemetry, !this.mongodb),
+    isRSAPI: !this.mongodb,
```
Comment (bietkul) [inline src/core/api/reactiveSearchv3Api.js]: Check it too.

File: __tests__/AppBaseClientTest.js [hunk 3]
```diff
@@ -1,96 +1,96 @@
-const appbase = require('..');
+const appbase = require("..");
```
Comment (bietkul) [inline __tests__/AppBaseClientTest.js:1]: Why do you have formatting changes?

**Review by bietkul — CHANGES_REQUESTED** [review]

File: src/core/index.js [hunk 4]
```diff
@@ -18,25 +20,45 @@ function AppBase(config) {
...
-  // Parse url
-  if (url.slice(-1) === '/') {
-    url = url.slice(0, -1);
-  }
+
   let credentials = auth || null;
+  if (!config.mongodb) {
+    url = host + path;
+    // Parse url
+    if (url.slice(-1) === '/') {
```
Comment (bietkul) [inline src/core/index.js]: URL parsing should happen irrespective of search BE.

**Review by bietkul — CHANGES_REQUESTED** [review]

File: src/core/index.js (hunk 4 above)
Comment (mohdashraf010897) [reply src/core/index.js]: Fixed here 👉  https://github.com/appbaseio/appbase-js/pull/84/commits/314651a097b981e8fef8bf9cb083dcfcb3faafa6

File: src/core/api/reactiveSearchApi.js (hunk 1 above)
Comment (mohdashraf010897) [reply src/core/api/reactiveSearchApi.js]: https://www.loom.com/share/af739667196a46f3924736a4b311e4df

File: src/core/api/reactiveSearchv3Api.js (hunk 2 above)
Comment (mohdashraf010897) [reply src/core/api/reactiveSearchv3Api.js]: https://www.loom.com/share/af739667196a46f3924736a4b311e4df

File: src/core/api/reactiveSearchApi.js (hunk 1 above)
Comment (mohdashraf010897) [reply src/core/api/reactiveSearchApi.js]: @bietkul 
f5a18e7

File: src/core/api/reactiveSearchv3Api.js (hunk 2 above)
Comment (mohdashraf010897) [reply src/core/api/reactiveSearchv3Api.js]: @bietkul 
f5a18e7

Comment (bietkul) [reply src/core/api/reactiveSearchv3Api.js]:
```suggestion
    isRSAPI: true,
```

**Review by bietkul — CHANGES_REQUESTED** [review]

**Review by bietkul — APPROVED** [review]

**Pull request merged**